lib/datasource/remote/firebase/regions/create_region_use_case.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:seeds/datasource/remote/api/region_repository.dart';
+import 'package:seeds/datasource/remote/firebase/regions/firebase_database_regions_repository.dart';
+import 'package:seeds/datasource/remote/model/transaction_response.dart';
+import 'package:seeds/domain-shared/base_use_case.dart';
+
+class CreateRegionUseCase extends InputUseCase<TransactionResponse, _Input> {
+  final RegionRepository _regionRepository = RegionRepository();
+  final FirebaseDatabaseRegionsRepository _firebaseRegionsRepository = FirebaseDatabaseRegionsRepository();
+
+  static _Input input({
+    required String userAccount,
+    required String regionAccount,
+    required String title,
+    required String description,
+    required double latitude,
+    required double longitude,
+    required String imageUrl,
+  }) =>
+      _Input(
+        userAccount: userAccount,
+        regionAccount: regionAccount,
+        title: title,
+        description: description,
+        latitude: latitude,
+        longitude: longitude,
+        imageUrl: imageUrl,
+      );
+
+  @override
+  Future<Result<TransactionResponse>> run(_Input input) async {
+    /// Create a region on firebase
+    final Result<String> firebaseResult = await _firebaseRegionsRepository.createRegion(
```

Comment:
title is not going to be save on firebase?

Reply:
I am thinking that we do not duplicate data. So everything thats saved in chain, will not be saved on firebase. 
with exception of location because we need to search by locations using firebaase